internal/engine/synclet_build_and_deployer.go

```diff
@@ -33,44 +32,26 @@ func NewSyncletBuildAndDeployer(sm SyncletManager, kCli k8s.Client) *SyncletBuil
 }
 
 func (sbd *SyncletBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.RStore, specs []model.TargetSpec, stateSet store.BuildStateSet) (store.BuildResultSet, error) {
-	iTargets, kTargets := extractImageAndK8sTargets(specs)
-	if len(kTargets) != 1 || len(iTargets) != 1 {
-		return store.BuildResultSet{}, RedirectToNextBuilderf(
-			"SyncletBuildAndDeployer requires exactly one image spec and one k8s deploy spec")
-	}
-
-	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncletBuildAndDeployer-BuildAndDeploy")
-	span.SetTag("target", kTargets[0].Name)
-	defer span.Finish()
-
-	iTarget := iTargets[0]
-	state := stateSet[iTarget.ID()]
-	if err := sbd.canSyncletBuild(ctx, iTarget, state); err != nil {
-		return store.BuildResultSet{}, WrapRedirectToNextBuilder(err)
+	iTargets, err := extractImageTargetsForLiveUpdates(specs, stateSet)
+	if err != nil {
+		return store.BuildResultSet{}, err
 	}
 
-	return sbd.updateInCluster(ctx, iTarget, state)
-}
-
-// canSyncletBuild returns an error if we CAN'T build this manifest via the synclet
-func (sbd *SyncletBuildAndDeployer) canSyncletBuild(ctx context.Context,
-	image model.ImageTarget, state store.BuildState) error {
-
-	// SyncletBuildAndDeployer doesn't support initial build
-	if state.IsEmpty() {
-		return fmt.Errorf("prev. build state is empty; synclet build does not support initial deploy")
+	if len(iTargets) != 1 {
+		return store.BuildResultSet{}, RedirectToNextBuilderf("Synclet container builder needs exactly one image target")
 	}
 
-	if image.MaybeFastBuildInfo() == nil {
-		return fmt.Errorf("container build only supports FastBuilds")
+	iTarget := iTargets[0]
+	if !isImageDeployedToK8s(iTarget, extractK8sTargets(specs)) {
```

Comment:
ah so I see why we check `isImageDeployedToK8s` here -- but we DON'T check in the localContainerBuilder cuz that might be called with DockerCompose, yeah? BUT is there a risk of the localContainerBuilder trying to in-place update an _intermediate_ image?

Reply:
ya, the synclet won't work with docker-compose.

the local container builder should be ok as long as the BuildState for the image target is correct